File: MyFinance.Application/Dtos/CategoryResponseDto.cs
using System;
using MyFinance.Domain.Enums;

namespace MyFinance.Application.Dtos;

public class CategoryResponseDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;

    /// <summary>Classificação de cortabilidade dada pelo usuário; base do plano de corte da sugestão.</summary>
    public ExpenseNature Nature { get; set; }
}

File: MyFinance.Frontend/src/types/CategoryResponseDto.ts
// src/types/CategoryResponseDto.ts
import type { ExpenseNature } from './ExpenseNature';

export interface CategoryResponseDto {
    id: string;
    name: string;
    /** Classificação de cortabilidade dada pelo usuário; base do plano de corte da sugestão. */
    nature: ExpenseNature;
}
